src/utils/getWeb3.js

```diff
@@ -0,0 +1,25 @@
+import Web3 from "web3";
+
+const getWeb3 = new Promise((resolve, reject) => {
```

Comment:
Are you sure you want this to be a singleton? (Also, it doesn't look like `reject` is ever called)

Reply:
Good question - I hadn't considered this. This _does_ only ever need to resolved once, as it's just grabbing the web3 object off of the window or building its own if none is found. Is this a valid reason to leave it as is?

A higher order question in my head - are there reasons to keep a Promise like this as a singleton, or should I always be returning a new one? If so, what are the reasons to/not to?